src/PerformanceTests/Utils/AsyncCountdownEvent.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public sealed class AsyncCountdownEvent
```

Comment:
It is a mixture of https://github.com/StephenCleary/AsyncEx in combination with my own code.